[PyTorch] Profiler execution graph record tensor device

louisfeng opened the pull request:
Summary: During execution graph replay, we also want to know where the tensor is allocated to properly model the behavior. The device name is now captured inside the tensor tuple.

Test Plan:
buck build mode/dev-nosan caffe2/test:profiler --show-output
buck-out/gen/caffe2/test/profiler#binary.par test_profiler.TestExecutionGraph

Reviewed By: aaronenyeshi

Differential Revision: D38017717

aaronenyeshi reviewed (approved):
LGTM, reviewed internally

louisfeng committed:
[PyTorch] Profiler execution graph record tensor device (#82895)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/82895

During execution graph replay, we also want to know where the tensor is allocated to properly model the behavior. The device name is now captured inside the tensor tuple.

Test Plan:
buck build mode/dev-nosan caffe2/test:profiler --show-output
buck-out/gen/caffe2/test/profiler#binary.par test_profiler.TestExecutionGraph

Reviewed By: aaronenyeshi

Differential Revision: D38017717

fbshipit-source-id: 94b07743a39bbd2bf96968ab2f8caddbb5b34543
